src/features/vars-schema/hooks/vars-update-trigger.ts

```diff
@@ -9,14 +10,21 @@ interface CustomObject {
 	affectedDevices?: number[];
 }
 
+// See: https://www.postgresql.org/docs/13/protocol-message-formats.html#:~:text=The%20number%20of%20parameter%20values%20that%20follow
+const MAX_SAFE_SQL_BINDS = Math.pow(2, 16) - 1;
+
 // Env vars hooks
 const addEnvHooks = (
 	resource: string,
 	buildFilter: (
 		args: hooks.HookArgs & {
 			tx: Tx;
 		},
-	) => Promise<Filter | undefined>,
+	) => Promise<
+		| FilterObj
+		| [affectedIds: number[], filterBuilder: (ids: number[]) => FilterObj]
```

Comment:
Also tried to use differentiating props like
```suggestion
		| Filter
		| {
			affectedIds: number[],
			filterBuilder: (ids: number[]) => Filter
		  }
```
but `typeof filter.filterBuilder === 'function'` wasn't enough for TS to figure out the type of affectedIds properly as well :(